presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDictionarySelectiveReader.java

```diff
@@ -622,7 +622,15 @@ public void close()
     @Override
     public long getRetainedSizeInBytes()
     {
-        return INSTANCE_SIZE + sizeOf(currentDictionaryData) + sizeOf(values) + sizeOf(outputPositions);
+        return INSTANCE_SIZE +
+                sizeOf(values) +
+                sizeOf(outputPositions) +
+                sizeOf(stripeDictionaryData) +
```

Comment:
should this be currentDictionary?

Reply:
currentDictionary = stripeDictionary + rowDictionary

Reply:
@bhhari Thanks for reviewing the code! `currentDictionary` could be pointing to `stripeDictionaryData` or the `rowGroupDictionaryData` created locally in `openRowGroup()`. So maybe a more accurate way is to check if `currentDictionary` is the same as `stripeDictionaryData`, and if not, count both `stripeDictionaryData` and `currentDictionary`. How do you think?

Reply:
> currentDictionary = stripeDictionary + rowDictionary

@bhhari The dictionary is built either on stripeDictionaryData or the rowGroupDictionaryData created locally in openRowGroup(). For the first case, currentDictionaryData points to stripeDictionaryData and we just need to count the dictionary's retained size. For the second case, rowGroupDictionaryData contains  stripeDictionaryData and row group dictionary, and the dictionary is built on top of it. We need to count the dictionary's retained size, but stripeDictionaryData is also there and also need to be counted.
I updated the code and added some comments:
```
@Override
    public long getRetainedSizeInBytes()
    {
        return INSTANCE_SIZE +
                sizeOf(values) +
                sizeOf(outputPositions) +
                // dictionary could be built on stripeDictionaryData or the rowGroupDictionaryData created locally in openRowGroup(). For the first case, currentDictionaryData
                // points to stripeDictionaryData and we just need to count the dictionary's retained size. For the second case, we need to count both stripeDictionaryData and dictionary.
                dictionary.getRetainedSizeInBytes() +
                (stripeDictionaryData == currentDictionaryData ? 0 : sizeOf(stripeDictionaryData)) +
                sizeOf(stripeDictionaryOffsetVector) +
                sizeOf(stripeDictionaryLength) +
                sizeOf(rowGroupDictionaryLength) +
                sizeOf(evaluationStatus);
    }
```